packages/@aws-cdk/core/lib/stack-synthesizers/default-synthesizer.ts

```diff
@@ -337,6 +354,28 @@ export class DefaultStackSynthesizer implements IStackSynthesizer {
   }
 }
 
+class BootstraplessSynthesizer extends DefaultStackSynthesizer {
+  public addFileAsset(_asset: FileAssetSource): FileAssetLocation {
+    throw new Error('Cannot add assets to a Stack that uses the BootstraplessSynthesizer');
+  }
+
+  public addDockerImageAsset(_asset: DockerImageAssetSource): DockerImageAssetLocation {
+    throw new Error('Cannot add assets to a Stack that uses the BootstraplessSynthesizer');
+  }
+
+  public synthesizeStackArtifacts(session: ISynthesisSession): void {
+    assertBound(this.stack);
+
+    // do _not_ treat the template as an asset,
```

Comment:
Probably worth doing a size check on the template here as well.

Reply:
Can we do it as a quick follow-up to this issue? <bats eyelashes>